components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4569,4 +4570,40 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
+
+    public static String[] getRequestedOIDCScopes(String[] requestedScopes)
+            throws IdentityOAuthAdminException {
+        if (requestedScopes == null) {
+            throw new IdentityOAuthAdminException("Requested scopes are null");
+        }
+        String[] oidcScopes =
+                OAuth2ServiceComponentHolder.getInstance().getOAuthAdminService().getScopeNames();
+        if (oidcScopes == null) {
+            throw new IdentityOAuthAdminException("Error occurred during getting OIDC scopes");
+        }
+        Set<String> oidcScopeSet = new HashSet<>(Arrays.asList(oidcScopes));
```

Comment:
Wouldn't it be good to check whether the oidc scopes is empty before proceeding. If empty we can return

Reply:
fixed